### PR TITLE
Adds Terraform Installed checks

### PR DESCRIPTION
**Pull request by mcmarkj**
I'm not sure if this was the right way to go around this. Basically in a Terraform setup, minimum versions aren't enough. If one user has 0.13.2 and the other had 0.13.4 then that'll really play havoc with the state so everyone needs to be on the same version. However those versions aren't floats, so instead I've handled them as strings. 

I've created a new helper that'll compare floats or strings to validate they're the same. 

👨‍💻 Adds version checking for Terraform 
```
TerraformInstalled(required_version="0.12.9")
```

### CODE REVIEW

**Comment by alexburlton-sonocent** [comment]
Totally agree that for terraform we don't want a minimum version and instead want it pinned.

The old `terraform-doctor` script used `.terraform-version` to get the version string - could we make the daktari check support this as well? You could still pass a version by hand, but maybe also have a `use_tfenv` bool or something? Would mean we only have the one place to remember to update in our tf repo rather than two.

**Comment by mcmarkj** [comment]
> Totally agree that for terraform we don't want a minimum version and instead want it pinned.
> 
> The old `terraform-doctor` script used `.terraform-version` to get the version string - could we make the daktari check support this as well? You could still pass a version by hand, but maybe also have a `use_tfenv` bool or something? Would mean we only have the one place to remember to update in our tf repo rather than two.

Nice idea!

Reckon this does the trick?

```
    def __init__(self, required_version: Optional[str] = "", use_tfenv: Optional[bool] = False):
        self.required_version = required_version
        self.name = "terraform.installed"

        if use_tfenv:
            # Read the required tf-version from tfenv (i.e. .terraform-version files)
            tfenv_version = open(".terraform-version", "r").read().strip()
            logging.debug(f"Terraform version required from .terraform-version file is: {tfenv_version}")
            self.required_version = tfenv_version
```

**Comment by alexburlton-sonocent** [comment]
> > Totally agree that for terraform we don't want a minimum version and instead want it pinned.
> > The old `terraform-doctor` script used `.terraform-version` to get the version string - could we make the daktari check support this as well? You could still pass a version by hand, but maybe also have a `use_tfenv` bool or something? Would mean we only have the one place to remember to update in our tf repo rather than two.
> 
> Nice idea!
> 
> Reckon this does the trick?
> 
> ```
>     def __init__(self, required_version: Optional[str] = "", use_tfenv: Optional[bool] = False):
>         self.required_version = required_version
>         self.name = "terraform.installed"
> 
>         if use_tfenv:
>             # Read the required tf-version from tfenv (i.e. .terraform-version files)
>             tfenv_version = open(".terraform-version", "r").read().strip()
>             logging.debug(f"Terraform version required from .terraform-version file is: {tfenv_version}")
>             self.required_version = tfenv_version
> ```

Yep! FYI, the types don't need to be `Optional[foo]` if you're defaulting to a value. `Optional` is needed if we want the variable to be `None` sometimes :smile:

File: daktari/check.py [hunk 1]
```diff
@@ -39,7 +39,21 @@ def verify(self, passed: bool, dualMessage: str) -> CheckResult:
         else:
             return self.failed(pattern.sub(" not ", dualMessage))
 
-    def validate_version(
+    def validate_required_version(
+        self, application: str, installed_version: Union[float, str, None], required_version: Union[float, str, None]
```
Comment (alexburlton-sonocent) [inline daktari/check.py:43]: I guess `required_version` can be `Union[float, str]` since we're not passing `None` ever?

**Review by alexburlton-sonocent — APPROVED** [review]
LGTM

**Pull request merged**